src/components/IndexTable/IndexTable.js

```diff
@@ -0,0 +1,72 @@
+import React, { Component } from 'react'
+import { Box, CheckBox, DataTable } from 'grommet'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+
+const RoundedBox = styled(Box)`
+  border-radius: 0.5em;
+`
+
+class IndexTable extends Component {
+  state = { checked: [], columns: [] }
+
+  onCheck = (e, value) => {
+    const { checked } = this.state;
+    if (e.target.checked) {
+      checked.push(value);
+      this.setState({ checked });
+    } else {
+      this.setState({ checked: checked.filter(item => item !== value) });
+    }
+  }
+
+  onCheckAll = e =>
+    this.setState({ checked: e.target.checked? this.props.data.map(datum => datum.id) : [] })
+
+  render() {
+    const { checked } = this.state;
+    const { columns, data } = this.props;
+
+    return (
+      <RoundedBox background='white' margin={{ vertical: 'small' }} pad='medium'>
+        <DataTable
+          columns={[
+            {
+              property: "checkbox",
+              render: datum => (
+                <CheckBox
+                  key={datum.name}
+                  checked={checked.indexOf(datum.id) !== -1}
+                  onChange={e => this.onCheck(e, datum.id)}
+                />
+              ),
+              header: (
+                <CheckBox
+                  checked={checked.length === data.length}
+                  indeterminate={
+                    checked.length > 0 && checked.length < data.length
+                  }
+                  onChange={this.onCheckAll}
+                />
```

Comment:
These checkboxes need labels for accessibility. There's a prop for `CheckBox` that can be used or you can use `htmlFor` and `id` to pair up a label. Maybe the workflow name is the label? It depends on what these checkboxes are supposed to do.